lib/app/endpoint/terminology.rb

```diff
@@ -86,43 +93,45 @@ def find_param(parameters, name)
         parameters.parameter.detect { |param| param.name == name }&.value
       end
 
-      def normalize_valueset_coding(parameters)
-        if find_param(parameters, 'coding')
-          find_param(parameters, 'coding').to_hash
-        elsif find_param(parameters, 'system') && find_param(parameters, 'code')
-          { 'code' => find_param(parameters, 'code'), 'system' => find_param(parameters, 'system') }
-        else
-          {}
-        end
-      end
-
-      def normalize_codesystem_coding(parameters, system_param = nil)
+      def normalize_coding(parameters, system_param = nil)
         if find_param(parameters, 'coding')
-          find_param(parameters, 'coding').to_hash
-        elsif (find_param(parameters, 'url') || system_param) && find_param(parameters, 'code')
-          system_param ||= find_param(parameters, 'url')
-          { 'code' => find_param(parameters, 'code'), 'system' => system_param }
+          [find_param(parameters, 'coding').to_hash]
+        elsif system_param && find_param(parameters, 'code')
+          [{ 'code' => find_param(parameters, 'code'), 'system' => system_param }]
+        elsif find_param(parameters, 'codeableConcept')
+          # NOTE: This branch doesn't wrap the response in an array because the 'coding' field on a CodeableConcept is already an array
+          find_param(parameters, 'codeableConcept').to_hash['coding']
         else
-          {}
+          []
```

Comment:
it seems like if no codes are passed then a `None of the codes included in the CodeableConcept are valid in the valueset` would result which could be misleading.  might not be worth fixing in this PR if it requires a lot of rework and at least it doesn't silently pass.